src/validateEmail.test.js

```diff
@@ -16,5 +16,61 @@ describe(`Function 'validateEmail':`, () => {
       .toBeTruthy();
   });
 
-  // write more tests here
+  it(`should return 'false' for an email without domain`, () => {
+    const result = validateEmail('test@');
+
+    expect(result).toBe(false);
+  });
+
+  it(`should return 'false' for an email without username`, () => {
+    const result = validateEmail('@example.com');
+
+    expect(result).toBe(false);
+  });
+
+  it(`should return 'false' for an email without @ symbol`, () => {
+    const result = validateEmail('testexample.com');
+
+    expect(result).toBe(false);
+  });
+
+  it(`should return 'false' for an email with invalid characters`, () => {
+    const result = validateEmail('test@example!com');
+
+    expect(result).toBe(false);
+  });
+
+  it(`should return 'false' for an email with multiple @ symbols`, () => {
+    const result = validateEmail('test@example@com');
+
+    expect(result).toBe(false);
+  });
+
+  it(`should return 'false' for an email with dot as the
+   first character`, () => {
+    const result = validateEmail('.test@example.com');
+
+    expect(result).toBe(false);
+  });
+
+  it(`should return 'false' for an email with consecutive 
+  dots in personal_info`, () => {
+    const result = validateEmail('test..example@example.com');
+
+    expect(result).toBe(false);
+  });
+
+  it(`should return 'false' for an email with invalid
+   characters in personal_info`, () => {
+    const result = validateEmail('test$example@example.com');
+
+    expect(result).toBe(false);
+  });
+
+  it(`should return 'false' for an email with not 
+  allowed characters`, () => {
+    const result = validateEmail('test!example@example.com');
+
+    expect(result).toBe(false);
+  });
 });
```

Comment:
Also add the tests to check the Non-latin input, input with - and _ , positive inputs for the dots that go not one after the other. Also a test, to check the instance of each not allowed character: ! $ % & ' * + / = ? ^ { | } ~, using your knowledge of Js and loops)